compiler/rustc_codegen_llvm/src/coverageinfo/mod.rs

```diff
@@ -50,8 +50,11 @@ impl<'ll, 'tcx> CrateCoverageContext<'ll, 'tcx> {
 
     /// LLVM use a temp value to record evaluated mcdc test vector of each decision, which is called condition bitmap.
     /// This value is named `mcdc.addr` (same as clang) and is a 32-bit integer.
-    fn try_get_mcdc_condition_bitmap(&self, instance: &Instance<'tcx>) -> Option<&'ll llvm::Value> {
-        self.mcdc_condition_bitmap_map.borrow().get(instance).copied()
+    fn try_get_mcdc_condition_bitmap(
+        &self,
+        instance: &Instance<'tcx>,
+    ) -> Option<Vec<&'ll llvm::Value>> {
+        self.mcdc_condition_bitmap_map.borrow().get(instance).cloned()
```

Comment:
This could receive `decision_depth` and just return `Option<&'ll llvm::Value>` so that no more clone (thus alloc) is needed.